### PR TITLE
Remove software

### LINKED ISSUE #892: Remove software from devices
We currently have `Software` as an optional field for some devices (e.g. detectors). As we're adding software to session and acquisition, we should probably remove it from the device.

### PR DESCRIPTION
**Pull request by micahwoodard**
fixes #892 
- removed recording_software from Detector class 
- removed encoder_firmware from Disc class
-  removed software  and script from StimulusEpoch

### CODE REVIEW

File: src/aind_data_schema/core/session.py [hunk 1]
```diff
@@ -500,16 +500,6 @@ class StimulusEpoch(AindModel):
     )
     stimulus_name: str = Field(..., title="Stimulus name")
     session_number: Optional[int] = Field(default=None, title="Session number")
-    software: Optional[List[Software]] = Field(
```
Comment (saskiad) [inline src/aind_data_schema/core/session.py:503]: don't remove the software/script from the session. Thanks!

**Review by saskiad — CHANGES_REQUESTED** [review]
let's not remove software, etc, from the session class.

File: src/aind_data_schema/core/session.py (hunk 1 above)
Comment (micahwoodard) [reply src/aind_data_schema/core/session.py:496]: Okay added back in!

**Comment by jtyoung84** [comment]
Decided not to remove this software field

**Pull request closed without merging**